src/main/java/org/prebid/server/handler/CookieSyncHandler.java

```diff
@@ -81,66 +95,127 @@ public CookieSyncHandler(long defaultTimeout,
         final BiddersContext biddersContext = BiddersContext.builder().build();
 
-        return accountById(requestAccount, timeout)
-                .compose(account -> privacyEnforcementService.contextFromCookieSyncRequest(
-                                cookieSyncRequest, routingContext.request(), account, timeout)
-                        .map(privacyContext -> CookieSyncContext.builder()
-                                .routingContext(routingContext)
-                                .uidsCookie(uidsCookie)
-                                .cookieSyncRequest(cookieSyncRequest)
-                                .usersyncMethodChooser(
-                                        UsersyncMethodChooser.from(cookieSyncRequest.getFilterSettings()))
-                                .biddersContext(biddersContext)
-                                .timeout(timeout)
-                                .account(account)
-                                .privacyContext(privacyContext)
-                                .debug(debug)
-                                .build()));
+        final CookieSyncContext cookieSyncContext = CookieSyncContext.builder()
+                .routingContext(routingContext)
+                .uidsCookie(uidsCookie)
+                .cookieSyncRequest(cookieSyncRequest)
+                .usersyncMethodChooser(UsersyncMethodChooser.from(cookieSyncRequest.getFilterSettings()))
+                .biddersContext(biddersContext)
+                .timeout(timeout)
+                .debug(debug)
+                .build();
+
+        return Future.succeededFuture(cookieSyncContext);
     }
 
-    private CookieSyncRequest parseRequest(RoutingContext routingContext) {
-        final Buffer body = routingContext.getBody();
-        if (body == null) {
-            throw new InvalidCookieSyncRequestException("Request has no body");
+    private ObjectNode normalizeRequest(Buffer body) {
+        final ObjectNode bodyAsObjectNode;
+        try {
+            bodyAsObjectNode = mapper.decodeValue(body, ObjectNode.class);
+        } catch (DecodeException e) {
+            logger.info(REQUEST_BODY_CANNOT_BE_PARSED, e);
+            throw new InvalidCookieSyncRequestException(REQUEST_BODY_CANNOT_BE_PARSED);
+        }
+
+        return normalizeGppSid(bodyAsObjectNode);
+    }
+
+    private static ObjectNode normalizeGppSid(ObjectNode requestNode) {
+        final JsonNode gppSidNode = requestNode.get(GPP_SID);
+        if (gppSidNode == null) {
+            return requestNode;
         }
 
+        if (!gppSidNode.isTextual()) {
+            logger.info(INVALID_GPP_SID);
+            throw new InvalidCookieSyncRequestException(INVALID_GPP_SID);
+        }
+
+        final List<Integer> gppSid = parseGppSid(gppSidNode.textValue());
+
+        final ArrayNode gppSidArrayNode = requestNode.putArray(GPP_SID);
+        gppSid.forEach(gppSidArrayNode::add);
+
+        return requestNode;
+    }
```

Comment:
Since you are trying to work with Lists from the very beginning of request processing, maybe this can be replaced with custom json deserializer?